smoke/tests/commit_test.go

```diff
@@ -0,0 +1,131 @@
+package tests
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/dragonflyoss/nydus/smoke/tests/tool"
+	"github.com/dragonflyoss/nydus/smoke/tests/tool/test"
+	"github.com/google/uuid"
+)
+
+type CommitTestSuite struct {
+	t                          *testing.T
+	testImage                  string
+	testCommittedImage         string
+	testContainerName          string
+	testCommittedContainerName string
+	snapshotter                string
+}
+
+func (c *CommitTestSuite) TestCommitContainer() {
+	ctx := tool.DefaultContext(c.t)
+	c.snapshotter = os.Getenv("SNAPSHOTTER")
+	if c.snapshotter == "" {
+		c.snapshotter = "nydus"
+	}
+
+	// choose test mode
+	mode := os.Getenv("COMMIT_MODE")
+	if mode == "" {
+		mode = "fs-version-6"
+	}
+	switch mode {
+	case "oci":
+	case "fs-version-5":
+		ctx.Build.FSVersion = "5"
+	case "fs-version-6":
+		ctx.Build.FSVersion = "6"
+	case "zran":
+		ctx.Build.OCIRef = true
+	default:
+		c.t.Fatalf("commit don't support %s mode", mode)
+	}
+
+	// prepare commit image
+	image := os.Getenv("COMMIT_TEST_IMAGE")
+	if image == "" {
+		image = "wordpress:6.1.1"
+	} else {
+		if !tool.SupportContainerImage(tool.ImageRepo(c.t, image)) {
+			c.t.Fatalf("commit don't support image " + image)
+		}
+	}
+	_, _ = c.prepareImage(c.t, ctx, image)
+
+	// run contaienr
+	c.testContainerName = uuid.NewString()
+	_ = tool.RunContainer(c.t, c.testImage, c.snapshotter, c.testContainerName)
+
+	// make some modifications in the read-write layer for commit action.
+	modifyCmd := fmt.Sprintf("echo 'This is Nydus commit' > commit && nerdctl cp commit %s:/root/", c.testContainerName)
+	tool.RunWithoutOutput(c.t, modifyCmd)
+
+	// commit container
+	commitCmd := fmt.Sprintf("%s commit --container %s --target %s", ctx.Binary.Nydusify, c.testCommittedContainerName, c.testCommittedImage)
+	tool.RunWithoutOutput(c.t, commitCmd)
+
+	// run committed container
+	_ = tool.RunContainer(c.t, c.testCommittedImage, c.snapshotter, c.testCommittedContainerName)
```

Comment:
Should also do `nerdctl cp %s:/root/commit commit` to check the file content. :)